openapi/src/parts/workspace.yaml

```diff
@@ -408,8 +408,6 @@ components:
             GCP Location to use for cloud-based resources. If omitted, the location of the source
             resource will be used.
           type: string
-        azureContext:
```

Comment:
I think this change shows up a mismatch between the expected model for BPM and the actual model.

WSM API is written assuming that the billing profile is _independent_ of the cloud platform. That one profile would hold the billing information for multiple clouds. Therefore, putting billing profile on the workspace object made sense.

BPM does not implement multi-cloud billing profiles. Since that is the case, the billing profile should not be attached to the workspace. It should be attached to the cloud context. If it were, then we would need the `azureContext` here to supply the billing profile.

Essentially for Azure the billing profile has become the context, so you are short-cutting it here. I am uneasy with that choice.

@aherbst-broad @rtitle

Reply:
It seems like this is an existing limitation of the clone workspace API though-- it didn't support providing a CloudContext, and instead an optional `AzureContext` was added on while BPM was in the process of being integrated.

If I am tracing through the [code](https://github.com/DataBiosphere/terra-workspace-manager/blob/a5e171f815d14b7a074d41b23c13e4792fafe218/service/src/main/java/bio/terra/workspace/service/workspace/flight/CreateGcpContextFlightV2.java#L51) properly, GCP workspace cloning always uses the spend profile obtained from the parent workspace and creates a GCP Cloud Context with it.

Passing a CloudContext to clone seems like it would be reasonable if we support cross-cloud cloning in the future. But that will require refactoring in both the GCP and Azure implementations, so I don't see that keeping AzureContext as an optional argument to clone (which currently is ignored if BPM is enabled) is helpful.

(Note also that I filled out the PR description)

Reply:
Ah, thanks for that explanation. I misunderstood.

I think there is still a mismatch that we should address, but this PR doesn't make it any worse.

Reply:
I agree there is a mismatch in expectations that we need to address-- we've been pretty hand-wavy about whether or not BPM profiles will support multiple clouds. It would be good to nail of this down before we start the migration of Rawls GCP workspaces to WSM/BPM.